extensions/wikia/Flags/controllers/FlagsController.class.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+use Flags\Models\Flag;
+use Flags\Models\FlagType;
+
+class FlagsController extends WikiaController {
+
+	private
+		$action,
+		$model,
+		$params,
+		$status = false;
+
+	public function getFlagsForPage() {
+		$this->skipRendering();
+		$this->getRequestParams();
+
+		if ( !isset( $this->params['pageId'] ) ) {
+			return null;
+		}
+
+		$this->model = new Flag();
+
+		return $this->model->getFlagsForPage( $this->params['wikiId'], $this->params['pageId'] );
+	}
+
+	public function getFlag() {
+
+	}
+
+	public function flagAction() {
```

Comment:
This method is performing a write action so must require a POST and verify a user token to protect against CSRF.